libgnucash/app-utils/gnc-option-impl.hpp

```diff
@@ -0,0 +1,1066 @@
+
+    GncOptionAccountSelValue(const char* section, const char* name,
+                          const char* key, const char* doc_string,
+                          GncOptionUIType ui_type,
+                          const Account* value) :
+        OptionClassifier{section, name, key, doc_string}, m_ui_type{ui_type},
+        m_value{const_cast<Account*>(value)},
+        m_default_value{const_cast<Account*>(value)}, m_allowed{} {}
+    GncOptionAccountSelValue(const char* section, const char* name,
+                          const char* key, const char* doc_string,
+                          GncOptionUIType ui_type,
+                          GncOptionAccountTypeList&& allowed) :
+        OptionClassifier{section, name, key, doc_string}, m_ui_type{ui_type},
+        m_value{}, m_default_value{}, m_allowed{std::move(allowed)} {}
+    GncOptionAccountSelValue(const char* section, const char* name,
+                          const char* key, const char* doc_string,
+                          GncOptionUIType ui_type,
+                          const Account* value,
+                          GncOptionAccountTypeList&& allowed) :
+        OptionClassifier{section, name, key, doc_string}, m_ui_type{ui_type},
+        m_value{}, m_default_value{}, m_allowed{std::move(allowed)} {
+            if (!validate(value))
+                throw std::invalid_argument("Supplied Value not in allowed set.");
+            m_value = const_cast<Account*>(value);
+            m_default_value = const_cast<Account*>(value);
+        }
+
+    const Account* get_value() const;
+    const Account* get_default_value() const;
+    bool validate (const Account* value) const;
+    void set_value (const Account* value) {
+        if (validate(value))
+            //throw!
+            m_value = const_cast<Account*>(value);
+    }
+    void set_default_value (const Account* value) {
+        if (validate(value))
+            //throw!
+            m_value = m_default_value = const_cast<Account*>(value);
+    }
+    GList* account_type_list() const noexcept;
+    void reset_default_value() { m_value = m_default_value; }
+    bool is_changed() const noexcept { return m_value != m_default_value; }
+    GncOptionUIType get_ui_type() const noexcept { return m_ui_type; }
+    void make_internal() { m_ui_type = GncOptionUIType::INTERNAL; }
+private:
+    GncOptionUIType m_ui_type;
+    Account* m_value;
+    Account* m_default_value;
+    GncOptionAccountTypeList m_allowed;
+};
+
+template<> inline std::ostream&
+operator<< <GncOptionAccountSelValue>(std::ostream& oss,
+                                       const GncOptionAccountSelValue& opt)
+{
+    auto value{opt.get_value()};
+    oss << qof_instance_to_string(QOF_INSTANCE(value));
+    return oss;
+}
+
+template<> inline std::istream&
+operator>> <GncOptionAccountSelValue>(std::istream& iss,
+                                   GncOptionAccountSelValue& opt)
+{
+    const Account* value;
```

Comment:
value maybe used uninitiated. Changed to const Account* value = nullptr;

Reply:
Thanks for all three.